### ISSUE TITLE
Let PD check regions before splitting regions

### ISSUE THREAD

**Issue by Yisaer** [pingcap/pd#2733]
### Describe your feature request related problem
Currently, for the splitting table cases, TiDB will notify TiKV to splitting the regions first, after the spllitting finished, the TiDB will inform PD to do the scattering job. However, in real workloads, during splitting, the status of the initial regions might be invalid from PD's perspective(non-replicated). So after splitting finished, the newly-created regions will also be invalid and PD will refuse to scatter these regions and wait coordinator to fix these regions which make the whole splitting process unstable in real workload.

### Describe the feature you'd like
Before TiDB splitting regions, let tidb notify PD to check the target regions and try to fix the invalid regions.

**Comment by disksing** [comment]
After notify, PD does not fix regions immediately. Do we need to check all regions become healthy before scatting them?

**Comment by Yisaer** [comment]
@disksing  Maybe we can check region before splitting table in TiDB. If the region count of this table is 1 and its replicas number is 4 (yet only need 3) at initial. After splitting table(eg: to 200 regions), there would be 200 regions needed to be fixed instead of 1 region for PD.

**Comment by Yisaer** [comment]
Of course, I think check and fixing regions before scatting is still worth to do.

**Comment by disksing** [comment]
> @disksing Maybe we can check region before splitting table in TiDB. If the region count of this table is 1 and its replicas number is 4 (yet only need 3) at initial. After splitting table(eg: to 200 regions), there would be 200 regions needed to be fixed instead of 1 region for PD.

Good point.
